### ISSUE TITLE
Inconsistent resource name casing in provider's schema

### ISSUE THREAD

**Issue by aureq** [pulumi/pulumi-azure-native#2365]
### What happened?

([context](https://pulumi.slack.com/archives/C02FXTZEZ6W/p1680934246894199))

In the Azure Native provider, I noticed that most of the resource names start with an upper case letter (`azure-native:advisor/v20160712preview:Suppression` => `Suppression`). This is nice because it correctly matches the name of resource in the TS/JS SDK.

However, for a few resources they seem to start with a lowercase letter (`azure-native:aadiam/v20200301preview:privateLinkForAzureAd` => `privateLinkForAzureAd`).

### Expected Behavior

I _think_ the resource name should start with an upper case letter (TitleCase) since the large majority follows the TitleCase.

### Steps to reproduce

Run the following command to list all resources with a lowercase resource name.

```bash
pulumi package get-schema azure-native@1.99.1 | jq -r '.resources | keys[]' | sed -n '/.*:.*:[a-z].*/p'
```

Here is the list of affected resources (68 resources)
```
azure-native:aadiam/v20200301:privateLinkForAzureAd
azure-native:aadiam/v20200301preview:privateLinkForAzureAd
azure-native:aadiam/v20200701preview:azureADMetric
azure-native:aadiam:azureADMetric
azure-native:aadiam:privateLinkForAzureAd
azure-native:azurestackhci/v20210701preview:galleryimageRetrieve
azure-native:azurestackhci/v20210701preview:networkinterfaceRetrieve
azure-native:azurestackhci/v20210701preview:virtualharddiskRetrieve
azure-native:azurestackhci/v20210701preview:virtualmachineRetrieve
azure-native:azurestackhci/v20210701preview:virtualnetworkRetrieve
azure-native:azurestackhci/v20210901preview:galleryimageRetrieve
azure-native:azurestackhci/v20210901preview:marketplacegalleryimage
azure-native:azurestackhci/v20210901preview:networkinterfaceRetrieve
azure-native:azurestackhci/v20210901preview:storagecontainerRetrieve
azure-native:azurestackhci/v20210901preview:virtualharddiskRetrieve
azure-native:azurestackhci/v20210901preview:virtualmachineRetrieve
azure-native:azurestackhci/v20210901preview:virtualnetworkRetrieve
azure-native:databricks/v20180401:vNetPeering
azure-native:databricks/v20210401preview:vNetPeering
azure-native:databricks/v20220401preview:vNetPeering
azure-native:databricks/v20230201:vNetPeering
azure-native:databricks:vNetPeering
azure-native:healthbot/v20201020:getbot
azure-native:healthbot/v20201020preview:getbot
azure-native:hybridcontainerservice/v20220501preview:agentPool
azure-native:hybridcontainerservice/v20220501preview:storageSpaceRetrieve
azure-native:hybridcontainerservice/v20220501preview:virtualNetworkRetrieve
azure-native:hybridcontainerservice:agentPool
azure-native:hybridcontainerservice:storageSpaceRetrieve
azure-native:hybridcontainerservice:virtualNetworkRetrieve
azure-native:insights/v20180601preview:guestDiagnosticsSetting
azure-native:insights:guestDiagnosticsSetting
azure-native:m365securityandcompliance/v20210325preview:privateLinkServicesForEDMUpload
azure-native:m365securityandcompliance/v20210325preview:privateLinkServicesForM365ComplianceCenter
azure-native:m365securityandcompliance/v20210325preview:privateLinkServicesForM365SecurityCenter
azure-native:m365securityandcompliance/v20210325preview:privateLinkServicesForMIPPolicySync
azure-native:m365securityandcompliance/v20210325preview:privateLinkServicesForO365ManagementActivityAPI
azure-native:m365securityandcompliance/v20210325preview:privateLinkServicesForSCCPowershell
azure-native:m365securityandcompliance:privateLinkServicesForEDMUpload
azure-native:m365securityandcompliance:privateLinkServicesForM365ComplianceCenter
azure-native:m365securityandcompliance:privateLinkServicesForM365SecurityCenter
azure-native:m365securityandcompliance:privateLinkServicesForMIPPolicySync
azure-native:m365securityandcompliance:privateLinkServicesForO365ManagementActivityAPI
azure-native:m365securityandcompliance:privateLinkServicesForSCCPowershell
azure-native:securityandcompliance/v20210111:privateLinkServicesForEDMUpload
azure-native:securityandcompliance/v20210111:privateLinkServicesForM365ComplianceCenter
azure-native:securityandcompliance/v20210111:privateLinkServicesForM365SecurityCenter
azure-native:securityandcompliance/v20210111:privateLinkServicesForO365ManagementActivityAPI
azure-native:securityandcompliance/v20210111:privateLinkServicesForSCCPowershell
azure-native:securityandcompliance/v20210308:privateLinkServicesForEDMUpload
azure-native:securityandcompliance/v20210308:privateLinkServicesForM365ComplianceCenter
azure-native:securityandcompliance/v20210308:privateLinkServicesForM365SecurityCenter
azure-native:securityandcompliance/v20210308:privateLinkServicesForMIPPolicySync
azure-native:securityandcompliance/v20210308:privateLinkServicesForO365ManagementActivityAPI
azure-native:securityandcompliance/v20210308:privateLinkServicesForSCCPowershell
azure-native:securityandcompliance:privateLinkServicesForEDMUpload
azure-native:securityandcompliance:privateLinkServicesForM365ComplianceCenter
azure-native:securityandcompliance:privateLinkServicesForM365SecurityCenter
azure-native:securityandcompliance:privateLinkServicesForMIPPolicySync
azure-native:securityandcompliance:privateLinkServicesForO365ManagementActivityAPI
azure-native:securityandcompliance:privateLinkServicesForSCCPowershell
azure-native:storagecache/v20230301preview:amlFilesystem
azure-native:synapse/v20210401preview:kustoPool
azure-native:synapse:kustoPool
azure-native:workloads/v20211201preview:monitor
azure-native:workloads/v20221101preview:monitor
azure-native:workloads/v20230401:monitor
azure-native:workloads:monitor
```

### Output of `pulumi about`

```
CLI          
Version      3.62.0
Go Version   go1.20.2
Go Compiler  gc

Host     
OS       debian
Version  11.6
Arch     x86_64
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by kpitzen** [comment]
Hey @aureq ! Thank you for raising this- I expect that a change like this would likely be breaking, so I'll be marking it as such. As it doesn't currently prevent the provider from functioning properly (we correctly generate SDKs, and can create resources from this schema), I'm also marking this as an enhancement rather than as a bug.

Thanks again!

**Comment by mikhailshilkov** [comment]
Fixed in 2.0
